activemount.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+)
+
+type activeMount struct {
+	UsageCount int
+}
+
+// activateVolume checks if the volume that has been requested to be mounted (as in docker volume mounting)
+// actually requires to be mounted (as an overlay fs mount). For that purpose check if other containers
+// have already mounted the volume (by reading in `activemountsdir`). It is also possible that the volume
+// has already been been mounted by the same container (when doing a `docker cp` while the container is running),
+// in that case the file named with the `requestId` will contain the number of times the container has
+// been requested the volume to be mounted. That number will be increased each time `activateVolume` is
+// called and decreased on `deactivateVolume`.
+// Parameters:
+//
+//	requestName: Name of the volume to be mounted
+//	requestID: ID of the container requesting the mount
+//	activemountsdir: Folder where Docker-On-Top mounts are tracked.
+//
+// Return:
+//
+//	doMountFs: Returns true if the request requires the filesystem to be mounted, false if not.
+//	err: If the function encountered an error, the error itself, nil if everything went right.
+func (d *DockerOnTop) activateVolume(requestName string, requestId string, activemountsdir lockedFile) (bool, error) {
+	var doMountFs bool
+
+	_, readDirErr := activemountsdir.ReadDir(1) // Check if there are any files inside activemounts dir
+	if readDirErr == nil {
+		// There is something no need to mount the filesystem again
+		doMountFs = false
+	} else if errors.Is(readDirErr, io.EOF) {
+		// The directory is empty, mount the filesystem
+		doMountFs = true
+	} else {
+		return false, fmt.Errorf("failed to list activemounts/ %v", readDirErr)
+	}
+
+	var activeMountInfo activeMount
+	activemountFilePath := d.activemountsdir(requestName) + requestId
+	file, err := os.Open(activemountFilePath)
+
+	if err == nil {
+		// The file can exist from a previous mount when doing a docker cp on an already mounted container, no need to mount the filesystem again
+		payload, readErr := io.ReadAll(file)
+		closeErr := file.Close()
+		if readErr != nil {
+			return false, fmt.Errorf("active mount file %s has been opened but cannot be read %w", activemountFilePath, readErr)
+		} else if closeErr != nil {
+			return false, fmt.Errorf("active mount file %s has been opened but cannot be closed %w", activemountFilePath, readErr)
+		}
+		unmarshalErr := json.Unmarshal(payload, &activeMountInfo)
+		if unmarshalErr != nil {
```

Comment:
No big deal but looks a bit too verbose to me. Could be
```go
if err := json.Unmarshal(...); err != nil {
    do stuff
}
```

Reply:
If you don't have strong feelings about it I'll keep it the way it is with a single instruction per line, I find it better specially when debugging.